breacharbiter.go

```diff
@@ -462,8 +627,9 @@ func (b *breachArbiter) breachObserver(contract *lnwallet.LightningChannel,
 		//
 		// TODO(roasbeef): also notify utxoNursery, might've had
 		// outbound HTLC's in flight
-		go waitForChanToClose(uint32(closeInfo.SpendingHeight), b.notifier,
-			nil, chanPoint, closeInfo.SpenderTxHash, func() {
+		go waitForChanToClose(uint32(closeInfo.SpendingHeight),
```

Comment:
In this specific case, the call back is not guaranteed to be executed, so the program could hang indefinitely during a shutdown. Lalou discovered this in the server/peer PR. Down the line it may be useful to add some sort of synchronization to these sorts of methods, so that it can signal the mutex regardless of whether the callback executed.

It's also somewhat unclear in this case that the breach arbiter should necessarily be responsible for this go routine, IIRC @Roasbeef pointed out that maybe the notifier should be responsible? For now the one is unsupervised, but we should find a loving home for it in the future.